### PR TITLE
Update library dependencies and remove marquee auto repeat

### PR DESCRIPTION
**Pull request by KevinGitonga**
Update library dependencies and update code to prevent auto repeat marquee loop once complete.
  I have update this library with the following changes.
- Update all dependencies to make it in sync with Oppia App project.
- Update MarquueView to make the Marquee run only once which is as required on our use case.
- Update magic value for "virtual width" to use double the device width instead.
- I have also commented out apply plugin: 'release' since this makes the App not to fully build.

## Sample texts used
Learn anything you want in an effective and enjoyable way, 
Share the experience and create up to 10 profiles,
Continue learning your lessons without internet connection,
User is able to download and delete content without Administrator PIN,
Share the experience and create up to 10 profiles,
All progress will be deleted and cannot be recovered.
Learn new math skills with stories that show you how to use them in your daily life,
The close button text of the dialog shown when a user's pre-release app has expired
Clicking on this will close the concept card.

## Default Virtual text width 2000
https://user-images.githubusercontent.com/20886444/188678386-9008783c-3290-4a9f-90bd-e618afc9850b.mp4

## Device width as virtual text width
https://user-images.githubusercontent.com/20886444/188678795-4e7ccb7b-9e80-4b36-8700-93af138c6f64.mp4

## Device width x2 as virtual text width
https://user-images.githubusercontent.com/20886444/188679871-183e8ab0-2416-4862-afc4-8dae12c954a0.mp4

## Text width as virtual text width
https://user-images.githubusercontent.com/20886444/188680397-5ccfb635-b269-404a-a05f-531a897e36bf.mp4

### CODE REVIEW

**Comment by KevinGitonga** [comment]
@BenHenning I have made changes to this sdk to make it usable for our use case. PTAL at the PR comment for more info on changes made and advice.

**Comment by seanlip** [comment]
@BenHenning PTAL

File: build.gradle [hunk 1]
```diff
@@ -1,18 +1,20 @@
...
 
     dependencies {
-        classpath 'com.android.tools.build:gradle:1.1.3'
+        classpath 'com.android.tools.build:gradle:3.5.0'
         classpath 'com.github.townsfolk:gradle-release:1.2'
     }
 }
 
-apply plugin: 'release'
+//apply plugin: 'release'
```
Comment (BenHenning) [inline build.gradle]: You can probably outright delete this.

File: library/src/main/java/asia/ivity/android/marqueeview/MarqueeView.java [hunk 2]
```diff
@@ -42,6 +41,9 @@ public class MarqueeView extends LinearLayout {
 
     private float mTextDifference;
 
+    //Change from virtual width to use device width
```
Comment (BenHenning) [inline library/src/main/java/asia/ivity/android/marqueeview/MarqueeView.java]: Comment is probably not needed (as it's more explaining why the change was made than what the code is for).

File: library/src/main/java/asia/ivity/android/marqueeview/MarqueeView.java [hunk 3]
```diff
@@ -269,7 +274,6 @@ public void onAnimationEnd(Animation animation) {
                     reset();
                     return;
                 }
-                startTextFieldAnimation();
```
Comment (BenHenning) [inline library/src/main/java/asia/ivity/android/marqueeview/MarqueeView.java:272]: Why this change?

**Review by BenHenning — CHANGES_REQUESTED** [review]
Thanks @KevinGitonga! Left two thoughts--PTAL.

File: build.gradle (hunk 1 above)
Comment (KevinGitonga) [reply build.gradle]: Changes applied.

File: library/src/main/java/asia/ivity/android/marqueeview/MarqueeView.java (hunk 2 above)
Comment (KevinGitonga) [reply library/src/main/java/asia/ivity/android/marqueeview/MarqueeView.java]: Noted.

File: library/src/main/java/asia/ivity/android/marqueeview/MarqueeView.java (hunk 3 above)
Comment (KevinGitonga) [reply library/src/main/java/asia/ivity/android/marqueeview/MarqueeView.java:272]: This makes sure once the Animation comes to an end and the cycle showing the full text and then reverting the text back to normal state runs once and does not repeat. By default the library repeats the Animation again once it comes to an end in a forever loop manner.

**Comment by KevinGitonga** [comment]
@BenHenning I have noted that if the device width is small e.g 720px for my case the text will not fully expand if it's very long so i doubled this to get more larger value this makes sense since the developer used dummy value 2000 as virtual width which allows for more room to expand the text. PTAL.

File: library/src/main/java/asia/ivity/android/marqueeview/MarqueeView.java [hunk 4]
```diff
@@ -141,6 +142,9 @@ private void init() {
         mPaint.setStrokeCap(Paint.Cap.ROUND);
 
         mInterpolator = new LinearInterpolator();
+
+        DisplayMetrics displayMetrics = getResources().getDisplayMetrics();
+        textViewVirtualWidth = displayMetrics.widthPixels * 2;
```
Comment (BenHenning) [inline library/src/main/java/asia/ivity/android/marqueeview/MarqueeView.java]: Why was the ``* 2`` needed?

Edit: I just saw your follow-up comment. Replying to it via this comment:

> @BenHenning I have noted that if the device width is small e.g 720px for my case the text will not fully expand if it's very long so i doubled this to get more larger value this makes sense since the developer used dummy value 2000 as virtual width which allows for more room to expand the text. PTAL.

This feels almost as arbitrary as the original 2000 value. How do we know that the "* 2" will work for all devices? I think it would also help for you to upload brief videos showing each of the three cases to make it clearer what behavior is changing with these values.

**Review by BenHenning — COMMENTED** [review]
Thanks @KevinGitonga! Just had one follow-up comment--PTAL.

**Comment by KevinGitonga** [comment]
> @BenHenning I have noted that if the device width is small e.g 720px for my case the text will not fully expand if it's very long so i doubled this to get more larger value this makes sense since the developer used dummy value 2000 as virtual width which allows for more room to expand the text. PTAL.

@BenHenning have quoted on above comment as reason for using double width as virtualTextWidth.

**Comment by KevinGitonga** [comment]
@BenHenning i have updated virtualTextWidth to use the actual width measurement for the actual text being animated. This makes better sense that using the device's width PTAL and advice.

File: library/src/main/java/asia/ivity/android/marqueeview/MarqueeView.java (hunk 3 above)
Comment (BenHenning) [reply library/src/main/java/asia/ivity/android/marqueeview/MarqueeView.java:272]: So it seems that this was reverted--was that intentional?

**Comment by BenHenning** [comment]
FYI @KevinGitonga I reopened an earlier conversation thread--PTAL.

File: library/src/main/java/asia/ivity/android/marqueeview/MarqueeView.java (hunk 4 above)
Comment (KevinGitonga) [reply library/src/main/java/asia/ivity/android/marqueeview/MarqueeView.java]: I have updated this to use the texts width instead of the device's width this makes sense and works well.

File: library/src/main/java/asia/ivity/android/marqueeview/MarqueeView.java (hunk 3 above)
Comment (KevinGitonga) [reply library/src/main/java/asia/ivity/android/marqueeview/MarqueeView.java:272]: Ooh reverting this change.

**Comment by KevinGitonga** [comment]
@BenHenning have reverted the change to prevent automatic repeat of animations.

File: library/src/main/java/asia/ivity/android/marqueeview/MarqueeView.java [hunk 5]
```diff
@@ -213,6 +213,7 @@ private void prepareAnimation() {
         mPaint.setTextSize(mTextField.getTextSize());
         mPaint.setTypeface(mTextField.getTypeface());
         float mTextWidth = mPaint.measureText(mTextField.getText().toString());
+        textViewVirtualWidth = (int) mTextWidth;
```
Comment (BenHenning) [inline library/src/main/java/asia/ivity/android/marqueeview/MarqueeView.java]: Nit: this is technically the actual width, not a virtual width. :) Probably worth updating to avoid potential future confusion.

**Review by BenHenning — COMMENTED** [review]
Thanks @KevinGitonga. Looking at the videos, it seems like each one is cutting off the text at least partly, but at different points. Would it be possible to change the texts to be English sentences and include them in your PR description to provide the context for analyzing correctness? It's hard to tell which solution actually works based on the current videos.

File: library/src/main/java/asia/ivity/android/marqueeview/MarqueeView.java (hunk 5 above)
Comment (KevinGitonga) [reply library/src/main/java/asia/ivity/android/marqueeview/MarqueeView.java]: Yeah @BenHenning this makes sense.

**Comment by KevinGitonga** [comment]
@BenHenning i have updated the text used in demo on PR and also uploaded english videos. Please note that while using the default 2000 and the text width as virtual width. The text is able to exapnd fully while the animation is playing.

**Review by BenHenning — APPROVED** [review]
> @BenHenning i have updated the text used in demo on PR and also uploaded english videos. Please note that while using the default 2000 and the text width as virtual width. The text is able to exapnd fully while the animation is playing.

Thanks @KevinGitonga. The updated videos really help. It seems like the only thing width changes is speed--is that correct?

Either way, this seems sensible so going ahead and approving/merging this.

**Comment by KevinGitonga** [comment]
> > @BenHenning i have updated the text used in demo on PR and also uploaded english videos. Please note that while using the default 2000 and the text width as virtual width. The text is able to exapnd fully while the animation is playing.
> 
> Thanks @KevinGitonga. The updated videos really help. It seems like the only thing width changes is speed--is that correct?
> 
> Either way, this seems sensible so going ahead and approving/merging this.

Thanks @BenHenning speed can be set while initializing MarqueeView, This will affect the duration which is a multiplication of speed and the textDifference. The larger the speed the longer the duration will be hence the animation will be much slower.
